Microsoft.Toolkit.Uwp.UI.Controls/WrapPanel/WrapPanel.cs

```diff
@@ -179,42 +198,69 @@ protected override Size MeasureOverride(Size availableSize)
-
-            double currentV = 0;
-            foreach (var child in Children)
+            if (Children.Count > 0)
             {
-                var desiredMeasure = new UvMeasure(Orientation, child.DesiredSize.Width, child.DesiredSize.Height);
-                if (desiredMeasure.U == 0)
-                {
-                    continue; // if an item is collapsed, avoid adding the spacing
-                }
+                var parentMeasure = new UvMeasure(Orientation, finalSize.Width, finalSize.Height);
+                var spacingMeasure = new UvMeasure(Orientation, HorizontalSpacing, VerticalSpacing);
+                var paddingStart = new UvMeasure(Orientation, Padding.Left, Padding.Top);
+                var paddingEnd = new UvMeasure(Orientation, Padding.Right, Padding.Bottom);
+                var position = new UvMeasure(Orientation, Padding.Left, Padding.Top);
 
-                if ((desiredMeasure.U + position.U + paddingEnd.U) > parentMeasure.U)
+                double currentV = 0;
+                void arrange(UIElement child, bool isLast = false)
                 {
-                    // next row!
-                    position.U = paddingStart.U;
-                    position.V += currentV + spacingMeasure.V;
-                    currentV = 0;
+                    var desiredMeasure = new UvMeasure(Orientation, child.DesiredSize.Width, child.DesiredSize.Height);
+                    if (desiredMeasure.U == 0)
+                    {
+                        return; // if an item is collapsed, avoid adding the spacing
+                    }
+
+                    if ((desiredMeasure.U + position.U + paddingEnd.U) > parentMeasure.U)
+                    {
+                        // next row!
+                        position.U = paddingStart.U;
+                        position.V += currentV + spacingMeasure.V;
+                        currentV = 0;
+                    }
+
+                    // Stretch the last item to fill the available space
+                    if (isLast)
+                    {
+                        desiredMeasure.U = parentMeasure.U - position.U;
+                    }
+
+                    // place the item
+                    if (Orientation == Orientation.Horizontal)
+                    {
+                        child.Arrange(new Rect(position.U, position.V, desiredMeasure.U, desiredMeasure.V));
+                    }
+                    else
+                    {
+                        child.Arrange(new Rect(position.V, position.U, desiredMeasure.V, desiredMeasure.U));
+                    }
+
+                    // adjust the location for the next items
+                    position.U += desiredMeasure.U + spacingMeasure.U;
+                    currentV = Math.Max(desiredMeasure.V, currentV);
                 }
 
-                // place the item
-                if (Orientation == Orientation.Horizontal)
+                if (StretchLastChild)
```

Comment:
🦙 Think this whole block (if + else) can become this:
```
for (var i = 0; i < Children.Count - 1; i++) {
     arrange(Children[i]);
}

arrange(Children[Children.Count - 1], StretchLastChild);
```
eh?

Reply:
Beautiful :) I'll change it